### ISSUE TITLE
[keycloak-js] Expose didInitialize as a public method/property

### ISSUE THREAD

**Issue by atomovski** [keycloak/keycloak#27040]
### Description

There is currently no way for a consumer of the keycloak instance to check if it is initialized without either keeping a local state or error handling. 

It would be ideal if the client would expose the already existing property `didInitialize`. 

### Motivation

As per the description, an additional motivation would be for React apps that have `StrictMode` enabled in which case effects are run twice and consumers of the library can easily get into a state where they attempt the init flow twice due to race conditions. 

### Details

I believe the implementation would be as simple as:

```
declare class Keycloak {
// Other properties

         /**
	 * Is true if the client is initialized, false otherwise.
	 */
	initialized?: boolean;
```
And then either rename the property in the JS file or expose a getter, whatever makes more sense for the owners.

**Comment by atomovski** [comment]
@jonkoops that's to a certain degree what I initially did, but I'm building a multi-tenanted app and the client needs to:
- initialize with a default realm  (doable outside of the react app)
- rely on local state to use the last realm the user worked with (somewhat doable outside of the react app)
- dynamically switch between realms, think of switching between account on Gmail as a simplified example (in this case I have to essentially rely on the client init process being delayed/lazy)

In all of the above integrating keykcloak with a state management library (say zustand) becomes harder and more involved if the init process needs to sit outside the app, given it relies on app state and produces state that the app needs to react to. 

I am sure there are solutions to the above that don't rely on exposing the `didInitialize` property (which I am using as it is technically available but I'd rather have an official API) but I was wondering if there are downsides to officially exposing this property? 

IMHO given that the init process is part of the usage pattern, it is not unreasonable for consumers to want to know if the client is initialized. The internals of the client anyway rely on this flag so it needs to be reliable. So I guess there's little downside in exposing it.

**Comment by jonkoops** [comment]
Yeah, I have no issue exposing this as a property. If you want to take this on feel free to do so. While we're at it, let's use that state to throw an exception if `init()` is called more than once.

**Comment by maapteh** [comment]
Im also wondering why with:

`import type Keycloak from 'keycloak-js'`

Keycloak is not typed with the `didInitialize` field, while debugging the field is there :(

Our usecase is we want to conditionally render it, when offline or failover on sso for example we do not wrap certain routes with keycloak. That means in our Layout I wrap the keycloak setup which gets loaded and unloaded.

What is the reason to not have it in the typing? @jonkoops maybe?
